File: engine/engine-model/src/main/scala/net/server/channel/packet/report/ReportWithChatPacket.scala
package net.server.channel.packet.report

class ReportWithChatPacket(private var _type: Int, private var _victim: String, private var _reason: Int, private var _description: String, private var _chatLog: String) extends BaseReportPacket(_type = _type, _victim = _victim, _reason = _reason, _description = _description) {
  def chatLog: String = _chatLog
}

File: engine/engine-model/src/main/java/net/server/channel/packet/report/ReportWithChatPacket.java
package net.server.channel.packet.report;

public class ReportWithChatPacket extends BaseReportPacket {
   private final String chatLog;

   public ReportWithChatPacket(Integer theType, String victim, Integer reason, String description, String chatLog) {
      super(theType, victim, reason, description);
      this.chatLog = chatLog;
   }

   public String chatLog() {
      return chatLog;
   }
}
